src/sensitivity_interface.jl

```diff
@@ -118,6 +195,12 @@ differencing, depending on the `autodiff` setting in the `AbstractSensitivityAlg
 Note that the keyword arguments are passed to the internal ODE solver for
```

Comment:
`then we assume `dgdp` is zero` is that still true @frankschae ?

Reply:
oh, no, it's not --  I thought it's better to assume that the general cost function ``g(u,p,t)`` may have a non-vanishing ``g_p``:
https://github.com/SciML/SciMLSensitivity.jl/blob/master/src/derivative_wrappers.jl#L855-L865

Reply:
Good, I thought that was fixed but was just double checking.

Do we specialize in concrete_solve dispatches to force this to zero? That probably still is a good performance optimization.

Reply:
How can I check if this is truly compiled away? 
We check if it's `nothing`
https://github.com/SciML/SciMLSensitivity.jl/blob/14fb96ea03ef70b224a5cb7a63b09c04ac82ebd6/src/adjoint_common.jl#L415
and from `concrete_solve`  we pass `dgdu_discrete` such that `dgdp_discrete` is set to nothing as the default, which gets wrapped into the `ReverseLossCallback` type.

Reply:
and we have  
```where {DG1, DG2, DG3, DG4, G}```
in the `AdjointODEProblem`s, e.g.,
https://github.com/SciML/SciMLSensitivity.jl/blob/14fb96ea03ef70b224a5cb7a63b09c04ac82ebd6/src/backsolve_adjoint.jl#L125

Reply:
that would compile away.